unlock-protocol-com/src/components/pages/State/index.tsx

```diff
@@ -0,0 +1,345 @@
+      {
+        name: 'Page Views',
+        data: [35, 41, 62, 42, 13, 18, 29, 37, 36, 51, 32, 35],
+      },
+      {
+        name: 'Total Visits',
+        data: [87, 57, 74, 99, 75, 38, 62, 47, 82, 56, 45, 47],
+      },
+    ],
+    options: {
+      chart: { zoom: { enabled: false } },
+      dataLabels: {
+        enabled: false,
+      },
+      markers: {
+        size: 0,
+        hover: {
+          sizeOffset: 6,
+        },
+      },
+      xaxis: {
+        categories: [
+          '01 Jan',
+          '02 Jan',
+          '03 Jan',
+          '04 Jan',
+          '05 Jan',
+          '06 Jan',
+          '07 Jan',
+          '08 Jan',
+          '09 Jan',
+          '10 Jan',
+          '11 Jan',
+          '12 Jan',
+        ],
+      },
+      yaxis: { show: false },
+      tooltip: {
+        y: [
+          {
+            title: {
+              formatter: function (val) {
+                return val + ' (mins)'
+              },
+            },
+          },
+          {
+            title: {
+              formatter: function (val) {
+                return val + ' per session'
+              },
+            },
+          },
+          {
+            title: {
+              formatter: function (val) {
+                return val
+              },
+            },
+          },
+        ],
+      },
+      grid: {
+        borderColor: '#f1f1f1',
+        xaxis: { lines: { show: true } },
+        yaxis: { lines: { show: true } },
+      },
+    },
+  }
+
+  return (
+    <div className="w-full h-80">
+      <ReactApexChart
+        options={chartOptions.options}
+        series={chartOptions.series}
+        type="line"
+        height={320}
+      />
+    </div>
+  )
+}
+
+function DateFilter({
+  filter,
+  setFilter,
+}: {
+  filter: string
+  setFilter: (value: string) => void
+}) {
+  return (
+    <div className="gap-4 flex flex-row items-center justify-center rounded-md bg-white p-2">
+      {filters.map((item, index) => (
+        <div
+          className="cursor-pointer"
+          onClick={() => setFilter(item)}
+          key={index}
+        >
+          <p
+            className={`text-gray font-lg px-3 py-1 ${
+              filter === item ? 'bg-black text-white rounded-md' : 'bg-white'
+            }`}
+          >
+            {item}
+          </p>
+        </div>
+      ))}
+    </div>
+  )
+}
+
+export function State() {
+  const [filter, setFilter] = useState('1Y')
+  const [isLoading, setIsLoading] = useState<boolean>(true)
+  const [gnpValues, setGNPValues] = useState<any[]>([])
+  const [subgraphData, setSubgraphData] = useState({})
+
+  useEffect(() => {
+    const run = async () => {
+      if (gnpValues.length > 0) {
+        const subgraphData = await querySubgraph(
+          'https://api.studio.thegraph.com/query/37457/test-unlock/v0.0.8/',
+          subgraphConfig
+        )
+        setSubgraphData(subgraphData.data)
+        console.log(subgraphData)
+      }
+    }
+    run()
+  }, [])
```

Comment:
use react-query if possible. That would cache and re-fetch in an interval on its own.

Reply:
resolved